File: Cap 08/Topico 8.3/Topico 8.3.1 e 8.3.2 juntos/mat2d.swift
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capítulo ..: 8 - Topico 8.3
 * 
 * Programa ..: ENTRADA E SAIDA EM MATRIZ 2D
 */

import Foundation

var NOTAS: [[Float]] = Array(repeating: Array(repeating: 0.0, count: 4), count: 8)
var I, J: Int

print("MATRIZ 2D - ENTRADA E SAIDA\n")
for I in 0...7 {
    print("Entre as notas do aluno \(I + 1):")
    for J in 0..<4 {
        print("Nota ==> \(J + 1): ", terminator: "")
        let input = readLine(strippingNewline: true)
        NOTAS[I][J] = Float(input!) ?? 0.0
    }
    print()
}
print("\nRELATORIO DE NOTAS\n")
print("Aluno Nota1 Nota2 Nota3 Nota4")
print("----- ----- ----- ----- -----")
for I in 0...7 {
    print(String(format: "%5d", I + 1), terminator: "")
    for J in 0..<4 {
        print(String(format: "%6.1f", NOTAS[I][J]), terminator: "")
    }
    print()
}

File: Cap 08/8.3 - Matrizes de Duas Dimensoes/8.3.1 e 8.3.2 juntos/mat2d.cs
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capítulo ..: 8 - Topico 8.3
 * 
 * Programa ..: ENTRADA E SAIDA EM MATRIZ 2D
*/

import 'dart:io';

void main() {
  var NOTAS = List<List<double>>.generate(8, (_) => List<double>.filled(4, 0.0));

  print('MATRIZ 2D - ENTRADA E SAIDA\n');
  for (var I = 0; I < 8; I++) {
    print('Entre as notas do aluno ${I + 1}:');
    for (var J = 0; J < 4; J++) {
      stdout.write('Nota ==> ${J + 1}: ');
      NOTAS[I][J] = double.parse(stdin.readLineSync()!);
    }
    print('');
  }

  print('\nRELATORIO DE NOTAS\n');
  print('Aluno Nota1 Nota2 Nota3 Nota4');
  print('----- ----- ----- ----- -----');
  for (var I = 0; I < 8; I++) {
    stdout.write('${I + 1}'.padLeft(5));
    for (var J = 0; J < 4; J++)
      stdout.write(' ${NOTAS[I][J].toStringAsFixed(1).padLeft(5)}');
    print('');
  }
}
